src/components/Main/Main.tsx

```diff
@@ -139,27 +163,54 @@ function Main() {
     }
   }, [autorunSimulation, debouncedRun, scenarioState, severity])
 
-  const [setScenarioToCustom] = useDebouncedCallback((newParams: AllParams) => {
+  const [setScenarioToCustom, cancelSetScenarioToCustom] = useDebouncedCallback((newParams: AllParams) => {
     // NOTE: deep object comparison!
-    if (!_.isEqual(allParams.population, newParams.population)) {
+    // Note: isEqual handles Date() objects while lodash.isEqual does not.
+    if (!isEqual(allParams.population, newParams.population)) {
       scenarioDispatch(setPopulationData({ data: newParams.population }))
     }
-    // NOTE: deep object comparison!
-    if (!_.isEqual(allParams.epidemiological, newParams.epidemiological)) {
+    if (!isEqual(allParams.epidemiological, newParams.epidemiological)) {
       scenarioDispatch(setEpidemiologicalData({ data: newParams.epidemiological }))
     }
-    // NOTE: deep object comparison!
-    if (!_.isEqual(allParams.simulation, newParams.simulation)) {
+    if (!isEqual(allParams.simulation, newParams.simulation)) {
       scenarioDispatch(setSimulationData({ data: newParams.simulation }))
     }
-    // NOTE: deep object comparison!
-    if (!_.isEqual(allParams.containment, newParams.containment)) {
+    if (!isEqual(allParams.containment, newParams.containment)) {
       scenarioDispatch(setContainmentData({ data: newParams.containment }))
     }
   }, 1000)
 
+  useEffect(() => {
+    onParamChange({ scenarioState, severity })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [scenarioState, severity])
+
+  useEffect(() => {
+    if (incomingParams) {
+      // Prior to update we must cancel any pending (debounced) call to set from
+      // form parameters. A re-render with prior scenario data can occur when
+      // switching views with multiple scenarios. The debounced call would
+      // change parameters for the new scenario.
+      cancelSetScenarioToCustom()
+      cancelDebouncedRun()
+
+      // Update state from incoming parameters.
+      scenarioDispatch(setScenarioData({ data: incomingParams.scenarioState.data }))
+      if (incomingParams.scenarioState.current !== CUSTOM_SCENARIO_NAME) {
+        scenarioDispatch(setScenario({ name: incomingParams.scenarioState.current }))
+      }
+      setSeverity(incomingParams.severity)
+    }
+  }, [incomingParams, cancelSetScenarioToCustom, cancelDebouncedRun])
+
+  useEffect(() => {
+    if (incomingResult) {
+      setResult(incomingResult)
+    }
+  }, [incomingResult])
+
```

Comment:
With these changes (more or less), the parent component <MultipleScenarios /> can control <Main />